src/odemis/driver/test/xt_client_test.py

```diff
@@ -281,6 +283,158 @@ def test_set_beam_shift(self):
+        print("X list is")
+        print(stigmator_values_x)
+        print("Y list is")
+        print(stigmator_values_y)
+        print("With a range of:")
+        print(self.microscope.stigmator_info()['range'])
+
+        # Test if an error is raised if an invalid detector is provided
+        with self.assertRaises(KeyError):
+            self.scanner.applyAutoStigmator("error_expected")
+        time.sleep(2.5)  # Give microscope/simulator the time to update the state
+        autostigmator_state = self.microscope.is_autostigmating(detector2ChannelName[detector])
+        self.assertEqual(autostigmator_state, False)  # Check if state remained unchanged
+
+    def test_apply_auto_contrast_brightness(self):
+        """
+        Test for the auto functionality of the apply_auto_contrast_brightness.
+        Only checks the returned state, the return object and if an error is raised for an incorrect input state.
+        """
+        for detector in detector2ChannelName:
+            auto_contrast_brightness_return = self.scanner.applyAutoContrastBrightness(detector)
+            time.sleep(2.5)  # Give microscope/simulator the time to update the state
+            auto_contrast_brightness_state = self.microscope.is_running_auto_contrast_brightness(detector2ChannelName[detector])
+            self.assertEqual(auto_contrast_brightness_state, True)
+            self.assertIsInstance(auto_contrast_brightness_return, ProgressiveFuture)
+
+            auto_contrast_brightness_return.cancel()
+            time.sleep(5.0)  # Give microscope/simulator the time to update the state
+            auto_contrast_brightness_state = self.microscope.is_running_auto_contrast_brightness(detector2ChannelName[detector])
+            self.assertEqual(auto_contrast_brightness_state, False)
+            self.assertIsInstance(auto_contrast_brightness_return, ProgressiveFuture)
+
+            self.scanner.applyAutoContrastBrightness(detector)
+            starting_time = time.time()
+            time.sleep(0.5)  # Give microscope/simulator the time to update the state
+            while self.microscope.is_running_auto_contrast_brightness(detector2ChannelName[detector]):
+                time.sleep(0.2)  # Wait until the autofocus is finished
+                if time.time() - starting_time > 30:
+                    break
+            # Check if the time to perform the autofocus is not to long
+            self.assertLess(time.time() - starting_time, 30, "Execution auto-contrast brightness is slow.")
+            # Line to inspect the execution time to update the expected time
+            print("Execution time was %s seconds " % (time.time() - starting_time))
+            auto_contrast_brightness_state = self.microscope.is_running_auto_contrast_brightness(detector2ChannelName[detector])
+            self.assertEqual(auto_contrast_brightness_state, False)
+
+        # Test if an error is raised if an invalid detector is provided
```

Comment:
missing: "or"

Reply:
Changed this to:
 > Test if an error is raised when an invalid detector is provided
 Is that clearer?